core/lib/zksync_core/src/sync_layer/fetcher.rs

```diff
@@ -94,15 +99,15 @@ pub struct MainNodeFetcher {
 }
 
 impl MainNodeFetcher {
-    pub async fn run(mut self) -> anyhow::Result<()> {
+    pub async fn run(mut self, stop_after_one_batch: bool) -> anyhow::Result<()> {
         tracing::info!(
             "Starting the fetcher routine. Initial miniblock: {}, initial l1 batch: {}",
             self.cursor.miniblock,
             self.cursor.l1_batch
         );
         // Run the main routine and reconnect upon the network errors.
         loop {
-            match self.run_inner().await {
+            match self.run_inner(stop_after_one_batch).await {
```

Comment:
Really stupid question: What is the purpose of stopping after one batch? It could be theoretically achieved without changing the `MainNodeFetcher` interface by sending a stop signal after observing an L1 batch getting processed.

Reply:
I don't like the idea of running those components 'for a while until certain conditions are met' that much, we can talk more about it using voice chat